### DISCUSSION TITLE
Is there anyway to disable log file writing?

### DISCUSSION THREAD

**Discussion by CharlinChen** [ggml-org/llama.cpp#3672] [Q&A] [answered]
The server was start by:
```
./server -m ./data/ggml-model-q8_0.gguf
```
The log file records too much content, and as the number of requests increases, the log file becomes larger and larger.
Is there any way to disable the output of log files

**Comment by KerfuffleV2** [comment, marked as answer]
You can compile with `LLAMA_DISABLE_LOGS=1` if you're using the `Makefile` (apparently there's no support when using cmake). Also, though it's not documented (which isn't ideal) there are some log-file related commandline options: https://github.com/ggerganov/llama.cpp/blob/f3b25e40438b3c8383caabf4e7b89863145a9f0e/common/log.h#L501-L525

(`--log-disable` is the one you're looking for.)

**Comment by misutoneko** [comment]
The compilation flags can also be given via the CFLAGS and CXXFLAGS environment variables, just add something like
`CFLAGS="-DLOG_DISABLE_LOGS=1" CXXFLAGS="-DLOG_DISABLE_LOGS=1"`
in front of your cmake command.
